### PR TITLE
Fix n+1

### PR DESCRIPTION
**Pull request by NurkaAmre**
### In this PR we implemented following features:
- **N+1 problem is solved when fetching all posts and their comments for a user.**
- **Used Capybara to write integration tests for each view of project.**
- **User index page:**

- [x] I can see the username of all other users.
- [x] I can see the profile picture for each user.
- [x] I can see the number of posts each user has written.
- [x] When I click on a user, I am redirected to that user's show page.

- **User show page:**
- [x] I can see the user's profile picture.
- [x] I can see the user's username.
- [x] I can see the number of posts the user has written.
- [x] I can See the user's bio.
- [x] I can see the user's first 3 posts.
- [x] I can see a button that lets me view all of a user's posts.
- [x] When I click a user's post, it redirects me to that post's show page.
- [x] When I click to see all posts, it redirects me to the user's post's index page.

- **User post index page:**
- [x] I can see the user's profile picture.
- [x] I can see the user's username.
- [x] I can see the number of posts the user has written.
- [x] I can see a post's title.
- [x] I can see some of the post's body.
- [x] I can see the first comments on a post.
- [x] I can see how many comments a post has.
- [x] I can see how many likes a post has.
- [x] I can see a section for pagination if there are more posts than fit on the view.
- [x] When I click on a post, it redirects me to that post's show page.

- **Post show page:**
- [x] I can see the post's title.
- [x] I can see who wrote the post.
- [x] I can see how many comments it has.
- [x] I can see how many likes it has.
- [x] I can see the post body.
- [x] I can see the username of each commentor.
- [x] I can see the comment each commentor left.

- ### Before:
Before posts comments

- ### Later:
After posts comments

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -160,6 +160,18 @@ Example:
...
+- Twitter: [@SajeelZafar6](https://twitter.com/SajeelZafar6)
+- LinkedIn: [Sajeel Zafar](https://www.linkedin.com/in/sajeelzafar/)
+
+👤 **Tashobya Israel**
+
+- GitHub: [@githubhandle](https://github.com/tashisrael)
+- Twitter: [@twitterhandle](https://twitter.com/tashisrael)
+- Linkedin: [@linkedinhandle](https://www.linkedin.com/in/tashobya-israel-6a66b0181/l)
+
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
```
Comment (t-yanick) [inline README.md:176]: - [ ] Great job working on this project. On line 239 of your ReadMe file is the MIT license link, but when I click on the link, I get erorr 404. Please kindly kix this as it is not professional to have broken links in our ReadMe
Screen Shot 2023-02-07 at 12 39 13 AM

File: spec/views/users/show.html.erb_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    end
+
+    it 'I can see the number of posts the user has written.' do
+      expect(page.body).to have_content(@user.PostCounter.to_s)
+    end
+
+    it 'I can see the user\'s bio' do
+      expect(page.body).to have_content(@user.bio)
+    end
+
```
Comment (t-yanick) [inline spec/views/users/show.html.erb_spec.rb:30]: - [ ] Good job adding the tests for the User show page. But you are missing the `test that asserts you can see the user's first 3 posts.` Kindly add this test.

File: spec/views/users/show.html.erb_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    it "I can see a button that lets me view all of a user's posts" do
+      expect(page.body).to have_content('See all posts')
+    end
+
+    it "When I click to see all posts button, it should redirects me to the user's post's index page" do
+      click_link('See all posts')
+      expect(page).to have_current_path(user_posts_path(@user))
+    end
```
Comment (t-yanick) [inline spec/views/users/show.html.erb_spec.rb:44]: - [ ] Still in the User show page tests, you are missing the `test that redirects to the post's show page when you click on a user's post`. Please kindly add this test as well.

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @NurkaAmre, @tashisrael and @Sajeelzafar, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Excellent and very descriptive Pull Request description.
:heavy_check_mark: Is a group project and all team members have contributed almost equally.
:heavy_check_mark:  All linter checks are passing

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Giving more descriptive names to your branch names. `feature01, feature02, ... feature09` are not quite descriptive

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @NurkaAmre, @Sajeelzafar and @tashisrael,

✔️ Good documentation

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**